src/Runner.Worker/JobExtension.cs

```diff
@@ -248,6 +249,19 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                     Trace.Info("Downloading actions");
                     var actionManager = HostContext.GetService<IActionManager>();
                     var prepareResult = await actionManager.PrepareActionsAsync(context, message.Steps);
+
+                    // add hook to preJobSteps
+                    var startedHookPath = Environment.GetEnvironmentVariable("ACTIONS_RUNNER_HOOK_JOB_STARTED");
+                    if (!string.IsNullOrEmpty(startedHookPath))
+                    {
+                        var hookStep = HostContext.CreateService<IManagedScriptStep>();
```

Comment:
should we just reuse `JobExtensionRunner`?

Reply:
done!